plugin/trino-hive/src/main/java/io/trino/plugin/hive/parquet/ParquetPageSourceFactory.java

```diff
@@ -329,6 +329,31 @@ public static Optional<org.apache.parquet.schema.Type> getColumnType(HiveColumnH
         if (baseColumnType.isEmpty() || column.getHiveColumnProjectionInfo().isEmpty()) {
             return baseColumnType;
         }
+
+        if (useParquetColumnNames && !column.getHiveColumnProjectionInfo().get().getSubfields().isEmpty()) {
+            try {
+                MessageType combinedPrunedTypes = null;
+                for (Subfield subfield : column.getHiveColumnProjectionInfo().get().getSubfields()) {
+                    MessageType prunedType = pruneRootTypeForSubfield(messageType, subfield);
+                    if (combinedPrunedTypes == null) {
+                        combinedPrunedTypes = prunedType;
+                    }
+                    else {
+                        combinedPrunedTypes = combinedPrunedTypes.union(prunedType);
+                    }
+                }
+                return Optional.ofNullable(combinedPrunedTypes) // Should never be null since subfields is non-empty.
```

Comment:
If `combinedPrunedTypes` should never be null, then you can just use `Optional.of(combinedPrunedTypes)`.

Reply:
Addressed in rev2